src/primitives/ray.jl

```diff
@@ -9,8 +9,8 @@ A ray originating at point `p`, pointed in direction `v`.
 It can be called as `r(t)` with `t > 0` to cast it at
 `p + t * v`.
 """
-struct Ray{C<:CRS,V<:Vec} <: Primitive{C}
-  p::Point{C}
+struct Ray{M<:𝔼,C<:CRS,V<:Vec} <: Primitive{M,C}
+  p::Point{M,C}
```

Comment:
please follow the same pattern of other geometries: p with Mp and Ray with M = E{Dim}